ext/xml/xml.c

```diff
@@ -663,15 +673,15 @@ void xml_startElementHandler(void *userData, const XML_Char *name, const XML_Cha
 			array_init(&tag);
 			array_init(&atr);
 
-			char *skipped_tag_name = SKIP_TAGSTART(ZSTR_VAL(tag_name));
-
-			xml_add_to_info(parser, skipped_tag_name);
+			zend_string *stripped_tag = xml_stripped_tag(tag_name, parser->toffset);
+			xml_add_to_info(parser, stripped_tag);
 
-			add_assoc_string(&tag, "tag", skipped_tag_name);
+			add_assoc_str(&tag, "tag", stripped_tag);
```

Comment:
It might be a good idea to add a comment to indicate that you are transferring the lifetime of the str to the tag array?